### DISCUSSION TITLE
Change "Last 30 days" to 28 to avoid data fluctuation

### DISCUSSION THREAD

**Discussion by muesliq** [plausible/analytics#2186] [Anything Else]
Web traffic is different on weekends and weekdays. Therefor the "Last 30 days" view fluctuates a lot, depending if that covers 4 or 5 weekends. Just look at these two charts of the same site that are offset by just 2 days, yet the unique user growth jumps by 5 percentage points:

<img width="1074" alt="Screenshot 2022-09-07 at 13 37 11" src="https://user-images.githubusercontent.com/1606618/188870186-0caa09a9-fa58-4e80-b3f9-5955c8cb01d4.png">

<hr />

<img width="1079" alt="Screenshot 2022-09-07 at 13 38 02" src="https://user-images.githubusercontent.com/1606618/188870220-ccc847b2-649e-41e4-9f6b-8623214e8480.png">

The rationale for 30 days is probably that that's how Google Analytics does it. 

But they do it wrong, too.

**Comment by muesliq** [comment]
Addendum: I would actually show **29 days** (the last 28 days + today) but just count the last 28 days, for the same reason (the incomplete data for today dilutes the comparison with the previous period).

This can be visually achieved by hatching the area for today. (The line is already correctly dashed.)

**Comment by Vrq** [comment]
Adding to the discussion:
Generally, "today" should not be used in the gain/loss % calculation because the current day never includes the traffic from the whole day. 

@metmarkosaric what is your take here?

**Reply by metmarkosaric** [reply]
thanks for the feedback @muesliq and @Vrq! it's something we haven't considered much at this stage but yeah at least it would be useful to remove today from this to make it more accurate

**Reply by muesliq** [reply]
If you want to improve reporting for relative numbers (and I would argue that you should want to - they are often more telling than absolute ones), this is what I would suggest:

1. Remove "today" from calculations but maybe keep showing it. (E.g. with a hatched area under the curve as a visual cue.)
2. Show 4 weeks instead of 30 days as outlined [above](https://github.com/plausible/analytics/discussions/2186)
3. Use "Visitor Growth" as a dimension in addition to "Visitors" [as outlined here](https://github.com/plausible/analytics/discussions/2097)

As for "industry standards" aka Google Analytics: They do use 30 days but "only" in the old parts of the application. The "Home" view gets it right already:

<img width="673" alt="Screenshot 2022-09-12 at 20 19 34" src="https://user-images.githubusercontent.com/1606618/189728471-d325179b-1846-43a0-977d-803812a03ea6.png">

**Comment by muesliq** [comment]
The following two screenshots are taken just two minutes apart, around midnight. They way those numbers are calculated now makes them pretty useless.
<img width="1104" alt="Screenshot 2022-09-22 at 23 56 36" src="https://user-images.githubusercontent.com/1606618/191947218-b5e83551-07ed-49fa-b328-0daea0b5b8fc.png">

<img width="1100" alt="Screenshot 2022-09-23 at 00 00 27" src="https://user-images.githubusercontent.com/1606618/191947232-2bb25b31-dbd1-42de-b40f-358b52070469.png">
